### DISCUSSION TITLE
error while runing mvn install

### DISCUSSION THREAD

**Discussion by AvishavTal** [devopvoid/webrtc-java#98] [Q&A]
After I execute mvn install I get this massege:
"Command: [C:\Users\avishav\IdeaProjects\webrtc-java\webrtc-jni\target\dependency\cmake\bin\cmake.exe, --build, C:\Users\avishav\IdeaProjects\webrtc-java\webrtc-jni\target\windows-x86_64, --target, install, --config, Release]
Directory: null
Environment: {USERDOMAIN_ROAMINGPROFILE=CLOUD, PROCESSOR_LEVEL=25, RegionCode=EMEA, VS140COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\, SESSIONNAME=Console, vs2022_install=C:\Program Files\Microsoft Visual Studio, ALLUSERSPROFILE=C:\ProgramData, PROCESSOR_ARCHITECTURE=AMD64, AMDRMSDKPATH=C:\Program Files\AMD\RyzenMasterSDK\, PSModulePath=C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules, SystemDrive=C:, MAVEN_HOME=C:\Program Files\maven\apache-maven-3.8.6, USERNAME=avishav, USERDNSDOMAIN=CLOUD.ISRCORP.CO.IL, ProgramFiles(x86)=C:\Program Files (x86), FPS_BROWSER_USER_PROFILE_STRING=Default, PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC, DriverData=C:\Windows\System32\Drivers\DriverData, IntelliJ IDEA Community Edition=C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2022.1.3\bin;, platformcode=KV, ProgramData=C:\ProgramData, ProgramW6432=C:\Program Files, HOMEPATH=\Users\avishav, PROCESSOR_IDENTIFIER=AMD64 Family 25 Model 80 Stepping 0, AuthenticAMD, ProgramFiles=C:\Program Files, PUBLIC=C:\Users\Public, windir=C:\WINDOWS, =::=::\, LOCALAPPDATA=C:\Users\avishav\AppData\Local, USERDOMAIN=CLOUD, FPS_BROWSER_APP_PROFILE_STRING=Internet Explorer, LOGONSERVER=\\ISR-DC1, OneDrive=C:\Users\avishav\OneDrive, APPDATA=C:\Users\avishav\AppData\Roaming, VBOX_MSI_INSTALL_PATH=C:\Program Files\Oracle\VirtualBox\, CommonProgramFiles=C:\Program Files\Common Files, Path=C:\Users\avishav\depot_tools;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\maven\apache-maven-3.8.6\bin\;C:\Program Files\Git\cmd;C:\Program Files (x86)\Microsoft SQL Server\160\DTS\Binn\;C:\Program Files\Azure Data Studio\bin;C:\Program Files\PuTTY\;C:\Program Files\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\dotnet\;C:\Program Files (x86)\IncrediBuild;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Users\avishav\AppData\Local\Microsoft\WindowsApps;;C:\Users\avishav\AppData\Local\Programs\Microsoft VS Code\bin;C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2022.1.3\bin;;C:\Program Files\Azure Data Studio\bin;C:\Users\avishav\.dotnet\tools, OS=Windows_NT, COMPUTERNAME=AVISHAVTAL, OnlineServices=Online Services, PROCESSOR_REVISION=5000, CommonProgramW6432=C:\Program Files\Common Files, ComSpec=C:\WINDOWS\system32\cmd.exe, DEPOT_TOOLS_WIN_TOOLCHAIN=0, SystemRoot=C:\WINDOWS, TEMP=C:\Users\avishav\AppData\Local\Temp, HOMEDRIVE=C:, USERPROFILE=C:\Users\avishav, TMP=C:\Users\avishav\AppData\Local\Temp, CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files, NUMBER_OF_PROCESSORS=16, IDEA_INITIAL_DIRECTORY=C:\Users\avishav\Desktop}
Exit code: 1"
What does that mean? 
How can I fix this?

**Comment by devopvoid** [comment]
`Directory: null` may be the problem. Not sure what directory it is. Seems to be related to CMake.

Is there more output? e.g. starting with:
`-- Building for: Visual Studio 16 2019` or any other VS version
until
`-- Build files have been written to:`

**Reply by AvishavTal** [reply]
Can't find the "Building for..." line.  A file containing all the output is attached. Thanks
[mvn install output.txt](https://github.com/devopvoid/webrtc-java/files/10031935/mvn.install.output.txt)

**Reply by devopvoid** [reply]
In your output.txt the lines 868 - 895 describe the error.

Line 872 (at the end): `win x86 environment.x86` ... not sure why that is, maybe you are running a x86 toolchain and not a x86_64.
Line 887: Exception: C:\Program Files\Microsoft Visual Studio\VC\vcvarsall.bat is missing - make sure VC++ tools are installed.

I would say: make sure VC++ tools are installed and try again. You can install them by changing the VS installation.

**Reply by AvishavTal** [reply]
I will try that. thanks
